### PR TITLE
Add some functions in gdi32-sys

### PR DESCRIPTION
**Pull request by varding**

### CODE REVIEW

File: src/wingdi.rs [hunk 1]
```diff
@@ -804,3 +840,259 @@ pub const FW_REGULAR: ::c_int = FW_NORMAL;
 pub const FW_DEMIBOLD: ::c_int = FW_SEMIBOLD;
 pub const FW_ULTRABOLD: ::c_int = FW_EXTRABOLD;
 pub const FW_BLACK: ::c_int = FW_HEAVY;
+
+pub type COLOR16 = ::c_ushort;
+
+#[repr(C)] #[derive(Clone, Copy, Debug, Default)]
```
Comment (retep998) [inline src/wingdi.rs]: Please no `Default`. It is easy enough to use `mem::zeroed` and I don't want compile times impacted by the extra codegen.

File: src/wingdi.rs [hunk 2]
```diff
@@ -804,3 +840,259 @@ pub const FW_REGULAR: ::c_int = FW_NORMAL;
...
+}
+
+pub type LPHANDLETABLE = *mut HANDLETABLE;
+pub type PHANDLETABLE = *mut HANDLETABLE;
+
+#[repr(C)] #[derive(Clone, Copy, Debug)]
+pub struct METARECORD {
+    pub rdSize: ::DWORD,
+    pub rdFunction: ::WORD,
+    pub rdParm: [::WORD;1],
```
Comment (retep998) [inline src/wingdi.rs]: Missing a space here

File: src/wingdi.rs [hunk 3]
```diff
@@ -804,3 +840,259 @@ pub const FW_REGULAR: ::c_int = FW_NORMAL;
...
+    pub nGlyphs: ::UINT,
+    pub nMaxFit: ::c_int,
+}
+
+pub type LPGCP_RESULTSW = *mut GCP_RESULTSW;
+
+#[repr(C)] #[derive(Clone, Copy, Debug)]
+pub struct FONTSIGNATURE {
+    pub fsUsb: [::DWORD;4],
+    pub fsCsb: [::DWORD;2],
```
Comment (retep998) [inline src/wingdi.rs]: Missing spaces here as well.

File: lib/user32-sys/src/lib.rs [hunk 4]
```diff
@@ -1080,3 +1101,23 @@ extern "system" {
...
+
+#[cfg(target_arch = "x86")]
+pub unsafe extern "system" fn SetWindowLongPtrA(hWnd: HWND, nIndex: c_int, dwNewLong: LONG_PTR) -> LONG_PTR {
+    SetWindowLongA(hWnd, nIndex, dwNewLong)
+}
+
+#[cfg(target_arch = "x86")]
+pub unsafe extern "system" fn SetWindowLongPtrW(hWnd: HWND, nIndex: c_int, dwNewLong: LONG_PTR) -> LONG_PTR {
+    SetWindowLongW(hWnd, nIndex, dwNewLong)
+}
```
Comment (retep998) [inline lib/user32-sys/src/lib.rs]: I'd prefer not to have these wrapper functions for now.

**Comment by retep998** [comment]
Do you think you could rebase this PR since there are some merge conflicts?

File: lib/user32-sys/src/lib.rs (hunk 4 above)
Comment (varding) [inline lib/user32-sys/src/lib.rs]: ok,I will delete code here

**Pull request merged**